bin/summarise_for_dram.py

```diff
@@ -0,0 +1,226 @@
+        for line in csv_reader:
+            # [
+            #   'ERZ12345_4_1',
+            #   _,
+            #   _,
+            #   'Pfam',
+            #   'PF12574',
+            #   '120 KDa Rickettsia surface antigen',
+            #   ...
+            (
+                protein_id,
+                _,
+                _,
+                analysis,
+                signature_accession,
+                signature_description,
+                *_
+            ) = line
+            contig = get_contig_id(protein_id)
+            if analysis.strip() == "Pfam":
+                pfams[contig].append(f"{signature_description} [{signature_accession}]")
+
+    # There are no pfams per contig that are lists, right?
+    for contig in pfams:
+        if isinstance(pfams[contig], list):
+            pfams[contig] = "; ".join(pfams[contig])
+
+    return pfams
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter
+    )
+    parser.add_argument(
+        "-f",
+        "--fasta",
+        type=str,
+        help="The faa with the proteins of the assembly.",
+    )
+    parser.add_argument(
+        "-i",
+        "--interproscan",
+        type=str,
+        help="InterProScan TSV output (not the summary)",
+    )
+    parser.add_argument("-k", "--ko-per-contig", type=str, help="KO per contig TSV")
+    parser.add_argument(
+        "-d",
+        "--dbcan-overview",
+        type=str,
+        help="run_DBCan overview output TSV file",
+    )
+    parser.add_argument("-p", "--prefix", type=str, help="TSV output prefix")
+
+    args = parser.parse_args()
+
+    assemblies_annotations = {}  # { assemblies_annotations[accessions] = set(contigs) }
+
+    kegg_annotations = args.ko_per_contig
+    ips_annotations = args.interproscan
+    dbcan_overview = args.dbcan_overview
+
+    # ----- CAZy ----- #
+
+    cazys = extract_cazy_families_from_dbcan(dbcan_overview)  # { cazys[contig] = cazy }
+
+    # ----- KEGG ----- #
+
+    keggs = extract_kegg_orthologs_annotations(
+        kegg_annotations
+    )  # { keggs[contig] = kegg }
+
+    # ----- IPS ----- #
+
+    pfams = extract_pfam_annotations(ips_annotations)  # { pfams[contig] = pfam }
+
+    # ----- Assemble tsv table ----- #
+
+    TABLE_HEADER = [
+        "",
+        "fasta",
+        "scaffold",
+        "gene_position",
+        "kegg_id",
+        "pfam_hits",
+        "cazy_id",
+    ]
+
+    # Collect the contig names
+    contigs = set()
+    with gzip.open(args.fasta, "rt") as fasta_file:
+        for line in fasta_file:
+            if line.startswith(">"):
+                # the assembly as this point needs to have been renamed
+                # So the contigs are named <prefix>_<incremental id>.. the prefix is
+                # usually the assembly ERZ accession
+                assembly, contig_id = line.strip().replace(">", "").split("_")
+                contigs.add(f"{assembly}_{contig_id}")
+
```

Comment:
Can't we pass the sample name from ${meta.id} to this script instead of catching the assembly name from the header? It seems a bit risky to get from the fasta header if it happens that '_' is part of the assembly name. Then the contig name could be the whole thing.

Reply:
One more observation here. The fasta file is the proteins of the assembly, right? That means that each entry corresponds to a protein, not to a contig. I'd like to see an example of the headers because I think you are catching gene/protein IDs instead of contig IDs. If the protein name is made from `<assembly_id>_<contig_id>_<protein incremental id>`, then this line will fail trying to extract two elements when there would be three.

Reply:
s.. I was mistakenly feeding the contigs 😮‍💨. I'll share an example.. because I am also confused about the headers and what the actual content of the file should look like

Reply:
I'd skip contigs without annotation. DRAM won't do a thing with them AFAIK